packages/yarnpkg-nm/sources/hoist.ts

```diff
@@ -438,6 +443,17 @@ const hoistTo = (tree: HoisterWorkTree, rootNodePath: Array<HoisterWorkTree>, ro
   return {anotherRoundNeeded, isGraphChanged};
 };
 
+const hasUnhoistedDependencies = (node: HoisterWorkTree): boolean => {
+  let result = false;
+  for (const [subName, subDependency] of node.dependencies) {
+    if (!node.peerNames.has(subName) && subDependency.ident !== node.ident) {
+      result = true;
+      break;
+    }
+  }
+  return result;
```

Comment:
Skip using variable `result`?
```suggestion
  for (const [subName, subDependency] of node.dependencies) {
    if (!node.peerNames.has(subName) && subDependency.ident !== node.ident) {
      return true
    }
  }
  return false;
```